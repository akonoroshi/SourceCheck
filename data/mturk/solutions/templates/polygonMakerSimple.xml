<project name="projects" app="Snap! 4.2, http://snap.berkeley.edu" version="1"><notes></notes><thumbnail>data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAKAAAAB4CAYAAAB1ovlvAAADg0lEQVR4Xu2cO6oiARREb4uBhgZiKoqCBn4QI92JrsPQbehCxIUIgrswMBM/g53NBAMvedcqjtCJCLeqzsGwi/f7/Q4+LJC0QIGASctztlwAAREhdQEETJ2f4wiIA6kLIGDq/BxHQBxIXQABU+fnOALiQOoCCJg6P8cREAdSF0DA1Pk5joA4kLoAAqbOz3EExIHUBRAwdX6OIyAOpC6AgKnzcxwBcSB1AQRMnZ/jCIgDqQsgYOr8HEdAHEhdAAFT5+c4AuJA6gIImDo/xxEQB1IXsBLwer1Go9FIHZTjP1vASsD9fh+XyyUmk0n0er2YzWZRq9V+tgi//tUFrAR8vV6x3W7jfD7H551Lj8cjptNpDIfD8hmNRlEUxa8OzLH/L2Al4Kfq7XaL9XodlUrlr+bP57P8bj6fx2q1ina7jRtfsICdgMfjMXa7XVSr1XLe+/1eyvb5B+z3+7FcLqNer3/B9ET4LGAl4Ol0is1mE61WK8bjcXQ6nVgsFtFsNqH9pQtYCXg4HGIwGES32/3SuYn17wJWAoJXbwEE1GNmlRgBrXDqlUFAPWZWiRHQCqdeGQTUY2aVGAGtcOqVQUA9ZlaJEdAKp14ZBNRjZpUYAa1w6pVBQD1mVokR0AqnXhkE1GNmlRgBrXDqlUFAPWZWiRHQCqdeGQTUY2aVGAGtcOqVQUA9ZlaJEdAKp14ZBNRjZpUYAa1w6pVBQD1mVokR0AqnXhkE1GNmlRgBrXDqlUFAPWZWiRHQCqdeGQTUY2aVGAGtcOqVQUA9ZlaJEdAKp14ZBNRjZpUYAa1w6pVBQD1mVokR0AqnXhkE1GNmlRgBrXDqlUFAPWZWiRHQCqdeGQTUY2aVGAGtcOqVQUA9ZlaJEdAKp14ZBNRjZpUYAa1w6pVBQD1mVokR0AqnXhkE1GNmlRgBrXDqlUFAPWZWiRHQCqdeGQTUY2aVGAGtcOqVQUA9ZlaJEdAKp14ZBNRjZpUYAa1w6pVBQD1mVokR0AqnXhkE1GNmlRgBrXDqlUFAPWZWiRHQCqdeGQTUY2aVGAGtcOqVQUA9ZlaJEdAKp14ZBNRjZpUYAa1w6pVBQD1mVokR0AqnXhkE1GNmlRgBrXDqlUFAPWZWiRHQCqdeGQTUY2aVGAGtcOqVQUA9ZlaJEdAKp14ZBNRjZpUYAa1w6pVBQD1mVokR0AqnXhkE1GNmlfgPzWQRt6Ej9NQAAAAASUVORK5CYII=</thumbnail><stage name="Stage" width="480" height="360" costume="0" tempo="60" threadsafe="false" lines="round" ternary="false" codify="false" inheritance="true" sublistIDs="false" scheduled="false" id="1"><pentrails>data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAeAAAAFoCAYAAACPNyggAAAOhUlEQVR4Xu3VwQkAAAjEMN1/abewn7jAQRC64wgQIECAAIF3gX1fNEiAAAECBAiMAHsCAgQIECAQCAhwgG6SAAECBAgIsB8gQIAAAQKBgAAH6CYJECBAgIAA+wECBAgQIBAICHCAbpIAAQIECAiwHyBAgAABAoGAAAfoJgkQIECAgAD7AQIECBAgEAgIcIBukgABAgQICLAfIECAAAECgYAAB+gmCRAgQICAAPsBAgQIECAQCAhwgG6SAAECBAgIsB8gQIAAAQKBgAAH6CYJECBAgIAA+wECBAgQIBAICHCAbpIAAQIECAiwHyBAgAABAoGAAAfoJgkQIECAgAD7AQIECBAgEAgIcIBukgABAgQICLAfIECAAAECgYAAB+gmCRAgQICAAPsBAgQIECAQCAhwgG6SAAECBAgIsB8gQIAAAQKBgAAH6CYJECBAgIAA+wECBAgQIBAICHCAbpIAAQIECAiwHyBAgAABAoGAAAfoJgkQIECAgAD7AQIECBAgEAgIcIBukgABAgQICLAfIECAAAECgYAAB+gmCRAgQICAAPsBAgQIECAQCAhwgG6SAAECBAgIsB8gQIAAAQKBgAAH6CYJECBAgIAA+wECBAgQIBAICHCAbpIAAQIECAiwHyBAgAABAoGAAAfoJgkQIECAgAD7AQIECBAgEAgIcIBukgABAgQICLAfIECAAAECgYAAB+gmCRAgQICAAPsBAgQIECAQCAhwgG6SAAECBAgIsB8gQIAAAQKBgAAH6CYJECBAgIAA+wECBAgQIBAICHCAbpIAAQIECAiwHyBAgAABAoGAAAfoJgkQIECAgAD7AQIECBAgEAgIcIBukgABAgQICLAfIECAAAECgYAAB+gmCRAgQICAAPsBAgQIECAQCAhwgG6SAAECBAgIsB8gQIAAAQKBgAAH6CYJECBAgIAA+wECBAgQIBAICHCAbpIAAQIECAiwHyBAgAABAoGAAAfoJgkQIECAgAD7AQIECBAgEAgIcIBukgABAgQICLAfIECAAAECgYAAB+gmCRAgQICAAPsBAgQIECAQCAhwgG6SAAECBAgIsB8gQIAAAQKBgAAH6CYJECBAgIAA+wECBAgQIBAICHCAbpIAAQIECAiwHyBAgAABAoGAAAfoJgkQIECAgAD7AQIECBAgEAgIcIBukgABAgQICLAfIECAAAECgYAAB+gmCRAgQICAAPsBAgQIECAQCAhwgG6SAAECBAgIsB8gQIAAAQKBgAAH6CYJECBAgIAA+wECBAgQIBAICHCAbpIAAQIECAiwHyBAgAABAoGAAAfoJgkQIECAgAD7AQIECBAgEAgIcIBukgABAgQICLAfIECAAAECgYAAB+gmCRAgQICAAPsBAgQIECAQCAhwgG6SAAECBAgIsB8gQIAAAQKBgAAH6CYJECBAgIAA+wECBAgQIBAICHCAbpIAAQIECAiwHyBAgAABAoGAAAfoJgkQIECAgAD7AQIECBAgEAgIcIBukgABAgQICLAfIECAAAECgYAAB+gmCRAgQICAAPsBAgQIECAQCAhwgG6SAAECBAgIsB8gQIAAAQKBgAAH6CYJECBAgIAA+wECBAgQIBAICHCAbpIAAQIECAiwHyBAgAABAoGAAAfoJgkQIECAgAD7AQIECBAgEAgIcIBukgABAgQICLAfIECAAAECgYAAB+gmCRAgQICAAPsBAgQIECAQCAhwgG6SAAECBAgIsB8gQIAAAQKBgAAH6CYJECBAgIAA+wECBAgQIBAICHCAbpIAAQIECAiwHyBAgAABAoGAAAfoJgkQIECAgAD7AQIECBAgEAgIcIBukgABAgQICLAfIECAAAECgYAAB+gmCRAgQICAAPsBAgQIECAQCAhwgG6SAAECBAgIsB8gQIAAAQKBgAAH6CYJECBAgIAA+wECBAgQIBAICHCAbpIAAQIECAiwHyBAgAABAoGAAAfoJgkQIECAgAD7AQIECBAgEAgIcIBukgABAgQICLAfIECAAAECgYAAB+gmCRAgQICAAPsBAgQIECAQCAhwgG6SAAECBAgIsB8gQIAAAQKBgAAH6CYJECBAgIAA+wECBAgQIBAICHCAbpIAAQIECAiwHyBAgAABAoGAAAfoJgkQIECAgAD7AQIECBAgEAgIcIBukgABAgQICLAfIECAAAECgYAAB+gmCRAgQICAAPsBAgQIECAQCAhwgG6SAAECBAgIsB8gQIAAAQKBgAAH6CYJECBAgIAA+wECBAgQIBAICHCAbpIAAQIECAiwHyBAgAABAoGAAAfoJgkQIECAgAD7AQIECBAgEAgIcIBukgABAgQICLAfIECAAAECgYAAB+gmCRAgQICAAPsBAgQIECAQCAhwgG6SAAECBAgIsB8gQIAAAQKBgAAH6CYJECBAgIAA+wECBAgQIBAICHCAbpIAAQIECAiwHyBAgAABAoGAAAfoJgkQIECAgAD7AQIECBAgEAgIcIBukgABAgQICLAfIECAAAECgYAAB+gmCRAgQICAAPsBAgQIECAQCAhwgG6SAAECBAgIsB8gQIAAAQKBgAAH6CYJECBAgIAA+wECBAgQIBAICHCAbpIAAQIECAiwHyBAgAABAoGAAAfoJgkQIECAgAD7AQIECBAgEAgIcIBukgABAgQICLAfIECAAAECgYAAB+gmCRAgQICAAPsBAgQIECAQCAhwgG6SAAECBAgIsB8gQIAAAQKBgAAH6CYJECBAgIAA+wECBAgQIBAICHCAbpIAAQIECAiwHyBAgAABAoGAAAfoJgkQIECAgAD7AQIECBAgEAgIcIBukgABAgQICLAfIECAAAECgYAAB+gmCRAgQICAAPsBAgQIECAQCAhwgG6SAAECBAgIsB8gQIAAAQKBgAAH6CYJECBAgIAA+wECBAgQIBAICHCAbpIAAQIECAiwHyBAgAABAoGAAAfoJgkQIECAgAD7AQIECBAgEAgIcIBukgABAgQICLAfIECAAAECgYAAB+gmCRAgQICAAPsBAgQIECAQCAhwgG6SAAECBAgIsB8gQIAAAQKBgAAH6CYJECBAgIAA+wECBAgQIBAICHCAbpIAAQIECAiwHyBAgAABAoGAAAfoJgkQIECAgAD7AQIECBAgEAgIcIBukgABAgQICLAfIECAAAECgYAAB+gmCRAgQICAAPsBAgQIECAQCAhwgG6SAAECBAgIsB8gQIAAAQKBgAAH6CYJECBAgIAA+wECBAgQIBAICHCAbpIAAQIECAiwHyBAgAABAoGAAAfoJgkQIECAgAD7AQIECBAgEAgIcIBukgABAgQICLAfIECAAAECgYAAB+gmCRAgQICAAPsBAgQIECAQCAhwgG6SAAECBAgIsB8gQIAAAQKBgAAH6CYJECBAgIAA+wECBAgQIBAICHCAbpIAAQIECAiwHyBAgAABAoGAAAfoJgkQIECAgAD7AQIECBAgEAgIcIBukgABAgQICLAfIECAAAECgYAAB+gmCRAgQICAAPsBAgQIECAQCAhwgG6SAAECBAgIsB8gQIAAAQKBgAAH6CYJECBAgIAA+wECBAgQIBAICHCAbpIAAQIECAiwHyBAgAABAoGAAAfoJgkQIECAgAD7AQIECBAgEAgIcIBukgABAgQICLAfIECAAAECgYAAB+gmCRAgQICAAPsBAgQIECAQCAhwgG6SAAECBAgIsB8gQIAAAQKBgAAH6CYJECBAgIAA+wECBAgQIBAICHCAbpIAAQIECAiwHyBAgAABAoGAAAfoJgkQIECAgAD7AQIECBAgEAgIcIBukgABAgQICLAfIECAAAECgYAAB+gmCRAgQICAAPsBAgQIECAQCAhwgG6SAAECBAgIsB8gQIAAAQKBgAAH6CYJECBAgIAA+wECBAgQIBAICHCAbpIAAQIECAiwHyBAgAABAoGAAAfoJgkQIECAgAD7AQIECBAgEAgIcIBukgABAgQICLAfIECAAAECgYAAB+gmCRAgQICAAPsBAgQIECAQCAhwgG6SAAECBAgIsB8gQIAAAQKBgAAH6CYJECBAgIAA+wECBAgQIBAICHCAbpIAAQIECAiwHyBAgAABAoGAAAfoJgkQIECAgAD7AQIECBAgEAgIcIBukgABAgQICLAfIECAAAECgYAAB+gmCRAgQICAAPsBAgQIECAQCAhwgG6SAAECBAgIsB8gQIAAAQKBgAAH6CYJECBAgIAA+wECBAgQIBAICHCAbpIAAQIECAiwHyBAgAABAoGAAAfoJgkQIECAgAD7AQIECBAgEAgIcIBukgABAgQICLAfIECAAAECgYAAB+gmCRAgQICAAPsBAgQIECAQCAhwgG6SAAECBAgIsB8gQIAAAQKBgAAH6CYJECBAgIAA+wECBAgQIBAICHCAbpIAAQIECAiwHyBAgAABAoGAAAfoJgkQIECAgAD7AQIECBAgEAgIcIBukgABAgQICLAfIECAAAECgYAAB+gmCRAgQICAAPsBAgQIECAQCAhwgG6SAAECBAgIsB8gQIAAAQKBgAAH6CYJECBAgIAA+wECBAgQIBAICHCAbpIAAQIECAiwHyBAgAABAoGAAAfoJgkQIECAgAD7AQIECBAgEAgIcIBukgABAgQICLAfIECAAAECgYAAB+gmCRAgQICAAPsBAgQIECAQCAhwgG6SAAECBAgIsB8gQIAAAQKBgAAH6CYJECBAgIAA+wECBAgQIBAICHCAbpIAAQIECAiwHyBAgAABAoGAAAfoJgkQIECAgAD7AQIECBAgEAgIcIBukgABAgQICLAfIECAAAECgYAAB+gmCRAgQICAAPsBAgQIECAQCAhwgG6SAAECBAgIsB8gQIAAAQKBgAAH6CYJECBAgIAA+wECBAgQIBAICHCAbpIAAQIECAiwHyBAgAABAoGAAAfoJgkQIECAgAD7AQIECBAgEAgIcIBukgABAgQIHLFxAWmhEwHPAAAAAElFTkSuQmCC</pentrails><costumes><list id="2"></list></costumes><sounds><list id="3"></list></sounds><variables></variables><blocks></blocks><scripts></scripts><sprites><sprite name="Sprite" idx="1" x="-56.838107018161054" y="87.94581568392181" heading="90" scale="1" rotation="1" draggable="true" costume="0" color="80,80,80" pen="tip" id="8"><costumes><list id="9"></list></costumes><sounds><list id="10"></list></sounds><blocks></blocks><variables></variables><scripts><script x="168" y="95"><block s="receiveGo"></block><block s="doAsk"><l>How many sides</l></block><block s="down"></block><block s="doRepeat"><block s="getLastAnswer"></block><script><block s="forward"><l>50</l></block><block s="turn"><block s="reportQuotient"><l>360</l><block s="getLastAnswer"></block></block></block></script></block></script></scripts></sprite></sprites></stage><hidden></hidden><headers></headers><code></code><blocks></blocks><variables></variables></project>