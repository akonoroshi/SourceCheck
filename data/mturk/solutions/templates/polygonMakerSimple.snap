snapshot {
    @defHint PEN_DOWN {
        :`Using the "pen down" block (under Pen) makes the sprite draw a trail behind it when it moves, which should happen before you draw a polygon.`
    }
    @defHint REPEAT {
        :`The "repeat" block (under Control) allows you to run the same code a fixed number of times, like drawing each side of a polygon.`
    }
    @defHint MOVE {
        :`The "move __ steps" block (under Motion) makes the sprite move forward some amount, like at each side of a polygon.`
    }
    @defHint TURN {
        :`The "turn __ degrees" block (under Motion) makes the sprite turn some degrees, like at each corner of a polygon.`
    }
    @defHint TURN_AMOUNT1 {
        :`The amount the sprite should turn after drawing each side is related to the number of sides.`
        priority(1)
    }
    @defHint TURN_AMOUNT2 {
        :`For a triangle (3 sides), the sprite should turn 120 degrees after each side. For a square (4 sides), it should be 90 degrees.`
        priority(2)
    }
    @defHint ASK {
        :`The "ask __ and wait" block (under Sensing) asks the user a question, like how many sides the polygon should have, and stores their response in the "answer" block (under Sensing).`
        priority(2)
    }
    @defHint SIDES {
        :`The "answer" block (under Sensing) should store how many sides the sprite should draw (in the "repeat" block), which also affects how much the sprite should turn after drawing each side.`
        priority(1)
    }
    stage:Stage {
        sprite:Sprite {
            script {
                @+optional { receiveGo }
                @anyOrder {
                    @+optional { clear }
                    @+optional { setHeading(literal:90) }
                    @+optional { gotoXY(literal:0, literal:0) }
                    @hint ASK
                    doAsk(literal:`How many sides`)
                    @hint PEN_DOWN
                    down
                }
                @hint REPEAT
                doRepeat(@hint ASK @hint SIDES getLastAnswer, script {
                    @defBlock DEGREES {
                        @hint TURN_AMOUNT1 @hint TURN_AMOUNT2
                        reportQuotient(literal:360,
                            @hint SIDES
                            getLastAnswer
                        )
                    }
                    @anyOrder {
                        @hint MOVE
                        forward(literal:50)
                        @or {
                            @hint TURN
                            turn(@block DEGREES)
                            @hint TURN
                            turnLeft(@block DEGREES)
                        }
                    }
                })
                @+optional { up }
            }
        }
    }
}